### PR TITLE
cbmc: remove `cmake` compiler designation.

### PR DESCRIPTION
**Pull request by NlightNFotis**
We now have incorporated this as part of an upstream change to make
compilation on osX more straightforward, so this is now redundant.

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Comment by NlightNFotis** [comment]
Hello again, I'm submitting another simplification to the formula for `cbmc` after a change we did upstream.

File: Formula/cbmc.rb [hunk 1]
```diff
@@ -29,8 +29,6 @@ class Cbmc < Formula
 
   def install
     args = []
```
Comment (carlocab) [inline Formula/cbmc.rb:31]:
```suggestion
```

File: Formula/cbmc.rb [hunk 2]
```diff
@@ -29,8 +29,6 @@ class Cbmc < Formula
...
-    # Workaround borrowed from https://github.com/diffblue/cbmc/issues/4956
-    args << "-DCMAKE_C_COMPILER=/usr/bin/clang" if OS.mac?
 
     system "cmake", "-S", ".", "-B", "build", *args, *std_cmake_args
```
Comment (carlocab) [inline Formula/cbmc.rb:33]:
```suggestion
    system "cmake", "-S", ".", "-B", "build", *std_cmake_args
```

**Review by carlocab — COMMENTED** [review]
Thanks, @NlightNFotis. We can remove a bit more now too.

**Review by carlocab — APPROVED** [review]
Thanks again!

**Pull request merged**